### ISSUE TITLE
Some mechanism to maybe define a feature policy

### ISSUE THREAD

**Issue by marcoscaceres** [w3c/manifest#798]
At TPAC, @aarongustafson, proposed: It might be nice to have some kind of "allow list" that would declare which APIs the installed web application can access after it is installed. This could potentially be a feature policy  (e.g., "allow": ["geolocation", "push"] ).

**Comment by dominickng** [comment]
Having a statically defined list of allowable permissions is useful in a number of different contexts. Some interesting questions:
 - should it also apply prior to installation (since the spec defines that a manifest applies to a document prior to installation as well as after)? There could be a disjointness if this isn't applied to sites that aren't installed

 - presumably this doesn't circumvent the on-use permission prompts, it simply acts as a filter on what permissions can be prompted for? This may or may not be left up to UAs to decide I guess

**Comment by fallaciousreasoning** [comment]
@beverloo noted that if we apply the manifest feature policy before install time we have to worry about races (manifest might not be loaded immediately).

**Comment by mgiuca** [comment]
Then I don't really see the point of this. If it's about adding more capabilities post install (e.g. consolidating or skipping permission prompts) then I could see value. If it's just about sites self-restricting what APIs are available, then there's no point unless it applies pre-install as well.

**Comment by aarongustafson** [comment]
> If it's about adding more capabilities post install (e.g. consolidating or skipping permission prompts) then I could see value. If it's just about sites self-restricting what APIs are available, then there's no point unless it applies pre-install as well.

It’s about a variety of things ;-) but most of them are either at or post installation. Also auditing. More coming in the explainer.

**Comment by marcoscaceres** [comment]
I'm also concerned about this. They could easily get out of sync between what the manifest declares and what the application actually does/needs. Possibly related:
https://marcosc.com/2014/12/why-manifests-and-permissions-dont-mix/

**Comment by aarongustafson** [comment]
Thank you for sharing this. I have ideas. More soon(ish).

**Comment by marcoscaceres** [comment]
I think I agree with @mgiuca (https://github.com/w3c/manifest/issues/798#issuecomment-533360172). There is a risk of the pre and post installation permission policies getting out of sync.  There are probably better mechanism to do this, like just using the Feature-Policy HTTP header.

**Comment by marcoscaceres** [comment]
@aarongustafson, any update on this from your side? We were waiting on some additional info - the original concerns about keeping things in sync remain, however.

**Comment by aarongustafson** [comment]
Closing this while we discuss https://github.com/MicrosoftEdge/MSEdgeExplainers/blob/master/InstallTimePermissionsPrompt/Explainer.md
